src/iden3comm/handlers/auth.ts

```diff
@@ -245,18 +268,49 @@ export class AuthHandler implements IAuthHandler {
       from: did.string(),
       to: authRequest.from
     };
+  }
 
-    const msgBytes = byteEncoder.encode(JSON.stringify(authResponse));
+  /**
+   * @inheritdoc IAuthHandler#parseAuthorizationRequest
+   */
+  async parseAuthorizationRequest(request: Uint8Array): Promise<AuthorizationRequestMessage> {
+    const { unpackedMessage: message } = await this._packerMgr.unpack(request);
+    const authRequest = message as unknown as AuthorizationRequestMessage;
+    if (message.type !== PROTOCOL_MESSAGE_TYPE.AUTHORIZATION_REQUEST_MESSAGE_TYPE) {
+      throw new Error('Invalid media type');
+    }
+    authRequest.body.scope = authRequest.body.scope || [];
+    return authRequest;
+  }
+
+  /**
+   * @inheritdoc IAuthHandler#handleAuthorizationRequest
+   */
+  async handleAuthorizationRequest(
+    did: DID,
+    request: Uint8Array,
+    opts?: AuthHandlerOptions
+  ): Promise<{
+    token: string;
+    authRequest: AuthorizationRequestMessage;
+    authResponse: AuthorizationResponseMessage;
+  }> {
+    const authRequest = await this.parseAuthorizationRequest(request);
+
+    const authResponse = await this.handleAuthRequest(authRequest, did, opts);
 
     const packerOpts =
-      opts.mediaType === MediaType.SignedMessage
+      opts?.mediaType === MediaType.SignedMessage
         ? opts.packerOptions
         : {
             provingMethodAlg: proving.provingMethodGroth16AuthV2Instance.methodAlg
           };
 
+    const msgBytes = byteEncoder.encode(JSON.stringify(authResponse));
+
+    const mediaType = opts?.mediaType ?? MediaType.ZKPMessage;
     const token = byteDecoder.decode(
-      await this._packerMgr.pack(opts.mediaType, msgBytes, {
+      await this._packerMgr.pack(mediaType, msgBytes, {
         senderDID: did,
         ...packerOpts
       })
```

Comment:
We could assign `const mediaType` before `const packerOpts` and then use `mediaType` instead of `opts?.mediaType` the same as we are doing in `const token`